Assets/Scripts/PlayerCharacter/PlayerGun.cs

```diff
@@ -0,0 +1,116 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Cosmobot
+{
+    public class PlayerGun : MonoBehaviour, DefaultInputActions.IPlayerGunActions
+    {
+
+        [SerializeField] private PlayerCamera playerCamera;
+        [SerializeField] private Transform cameraTransform;
+        [SerializeField] private float maxPickupRange; // relative to character, not camera
+        [SerializeField] private float carryHoldingDistance;
+        [SerializeField] private float holdingDistanceZoomMultiplier = 1f;
+        [SerializeField] private Vector3 carryPositionFirstPersonOffset;
+        [SerializeField] private Vector3 carryPositionThirdPersonOffset;
+        [SerializeField] private float carryForceMultiplier = 1f;
+        [SerializeField] private string playerLayerName = "Player";
+        [SerializeField] private string itemLayerName = "Item";
+
+        private bool carryingItem = false;
+        private Transform carriedItemTransform = null;
+        private Rigidbody carriedItemBody = null;
+
+        private Vector3 carryPosition;
+
+        private DefaultInputActions actions;
+
+        public void Update()
+        {
+            UpdateModel();
+        }
+
+        public void FixedUpdate()
+        {
+            var carryPosOffset = playerCamera.isFirstPerson ? carryPositionFirstPersonOffset : carryPositionThirdPersonOffset;
+            carryPosOffset = Quaternion.Euler(0, transform.rotation.eulerAngles.y, 0) * carryPosOffset;
+
+            var holdDistance = carryHoldingDistance * (playerCamera.isZoomed ? holdingDistanceZoomMultiplier : 1);
+
+            carryPosition = transform.position + cameraTransform.forward * holdDistance + carryPosOffset;
+            if (carryingItem)
+            {
+                var force = (carryPosition - carriedItemTransform.position) * carryForceMultiplier
+                    - carriedItemBody.velocity;
+                carriedItemBody.AddForce(force, ForceMode.VelocityChange);
+            }
+        }
```

Comment:
The whole code can be placed inside the `if (carryingItem)` block. Also, maintain consistent code style across your files. In the `Update` function, you only called a method to perform a simple task, while here you have placed everything directly within `FixedUpdate` instead of creating a separate method for this purpose.

```suggestion
        private void FixedUpdate()
        {
            CarryItem()
        }
                    
        private void CarryItem()
        {
            if (!carryingItem) return;
            
            var carryPosOffset = playerCamera.isFirstPerson ? carryPositionFirstPersonOffset : carryPositionThirdPersonOffset;
            carryPosOffset = Quaternion.Euler(0, transform.rotation.eulerAngles.y, 0) * carryPosOffset;

            var holdDistance = carryHoldingDistance * (playerCamera.isZoomed ? holdingDistanceZoomMultiplier : 1);

            carryPosition = transform.position + cameraTransform.forward * holdDistance + carryPosOffset;
            var force = (carryPosition - carriedItemTransform.position) * carryForceMultiplier
            - carriedItemBody.velocity;
            carriedItemBody.AddForce(force, ForceMode.VelocityChange);
        }
```